packages/@sanity/cli/src/actions/init-project/templates/nextjs/schemas/blog.ts

```diff
@@ -0,0 +1,444 @@
+      title: 'Main image',
+      type: 'image',
+      options: {
+        hotspot: true,
+      },
+    }),
+    defineField({
+      name: 'categories',
+      title: 'Categories',
+      type: 'array',
+      of: [{type: 'reference', to: {type: 'category'}}],
+    }),
+    defineField({
+      name: 'publishedAt',
+      title: 'Published at',
+      type: 'datetime',
+    }),
+    defineField({
+      name: 'body',
+      title: 'Body',
+      type: 'blockContent',
+    }),
+  ],
+
+  preview: {
+    select: {
+      title: 'title',
+      author: 'author.name',
+      media: 'mainImage',
+    },
+    prepare(selection) {
+      const {author} = selection
+      return {...selection, subtitle: author && \`by \${author}\`}
+    },
+  },
+})
+`
+
+const postJS = `export default post = {
+  name: 'post',
+  title: 'Post',
+  type: 'document',
+  fields: [
+    {
+      name: 'title',
+      title: 'Title',
+      type: 'string',
+    },
+    {
+      name: 'slug',
+      title: 'Slug',
+      type: 'slug',
+      options: {
+        source: 'title',
+        maxLength: 96,
+      },
+    },
+    {
+      name: 'author',
+      title: 'Author',
+      type: 'reference',
+      to: {type: 'author'},
+    },
+    {
+      name: 'mainImage',
+      title: 'Main image',
+      type: 'image',
+      options: {
+        hotspot: true,
+      },
+    },
+    {
+      name: 'categories',
+      title: 'Categories',
+      type: 'array',
+      of: [{type: 'reference', to: {type: 'category'}}],
+    },
+    {
+      name: 'publishedAt',
+      title: 'Published at',
+      type: 'datetime',
+    },
+    {
+      name: 'body',
+      title: 'Body',
+      type: 'blockContent',
+    },
+  ],
+
+  preview: {
+    select: {
+      title: 'title',
+      author: 'author.name',
+      media: 'mainImage',
+    },
+    prepare(selection) {
+      const {author} = selection
+      return {...selection, subtitle: author && \`by \${author}\`}
+    },
+  },
+})
+`
+
+// Schema definition
+
+export const blogSchemaTS = `import { type SchemaTypeDefinition } from 'sanity'
+
+import blockContent from './schemas/blockContent'
+import category from './schemas/category'
+import post from './schemas/post'
+import author from './schemas/author'
+
+export const schema: { types: SchemaTypeDefinition[] } = {
+  types: [post, author, category, blockContent],
+}
+`
+
+export const blogSchemaJS = `import blockContent from './schemas/blockContent'
+import category from './schemas/category'
+import post from './schemas/post'
+import author from './schemas/author'
+
+export const schema = {
+  types: [post, author, category, blockContent],
+}
+`
+
+export const blogSchemaFolder = (useTypeScript: boolean): Record<string, string> => {
+  return {
+    'author.': useTypeScript ? authorTS : authorJS,
+    'blockContent.': useTypeScript ? blockContentTS : blockContentJS,
+    'category.': useTypeScript ? categoryTS : categoryJS,
+    'post.': useTypeScript ? postTS : postJS,
+  }
+}
```

Comment:
This is the folder structure tree I referenced above. The key ends with `.` to indicate that it's a file and not a folder, then we append the `fileEnding` depending on whether you want TS or not.